### PR TITLE
Implement SMF writer.

### PR DESCRIPTION
**Pull request by wmedrano**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3,11 +3,15 @@
 #[macro_use]
 extern crate std;
 
+#[cfg(feature = "std")]
+extern crate integer_encoding;
```
Comment (wmedrano) [inline src/lib.rs:7]: This isn't used.

File: src/lib.rs [hunk 2]
```diff
@@ -3,11 +3,15 @@
...
+
 mod byte;
 mod cc;
 mod error;
 mod midi_message;
 mod note;
+pub mod smf;
```
Comment (wmedrano) [inline src/lib.rs:14]: Restrict this to `#[cfg(feature = "std")]`.

File: src/midi_message.rs [hunk 3]
```diff
@@ -158,6 +158,58 @@ impl<'a> MidiMessage<'a> {
         MidiMessage::try_from(bytes)
     }
 
+    /// Writes the data to `w`.
+    #[allow(clippy::range_plus_one)]
+    pub fn write<W: std::io::Write>(&self, w: &mut W) -> std::io::Result<usize> {
```
Comment (wmedrano) [inline src/midi_message.rs:163]: Rename to encode?

File: src/smf.rs [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+        for e in track.iter() {
+            bytes_written += e.encode(w)?;
+        }
+        Ok(bytes_written)
+    }
+}
+
+/// Meta events are non-MIDI data of various sorts consisting of a code and actual event data.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MetaEvent {
```
Comment (wmedrano) [inline src/smf.rs:109]: Finish all the variants

File: src/smf.rs [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+    SequenceNumber(String),
```
Comment (wmedrano) [inline src/smf.rs:110]: Investigate if this is really a string

**Comment by sourcebox** [comment]
I'm struggling a bit with this PR. There's no question that a SMF writer is very useful to have, but I would put that into a separate crate instead of merging it here.

**Comment by wmedrano** [comment]
I totally agree so I'm closing out this PR.

Also related, I've written a parser for SMF. https://github.com/Musical-Fruit/musical-fruit/blob/main/wmidi-smf/src/lib.rs
I have not open sourced it because it does not handle all SMF features and has not been thoroughly tested. but... I should release this while disclosing the previous statement as a caveat.

**Pull request closed without merging**